oap-server/server-library/library-server/src/main/java/org/apache/skywalking/oap/server/library/server/http/HTTPServer.java

```diff
@@ -70,6 +78,21 @@ public void initialize() {
             })
             .decorator(LoggingService.newDecorator());
 
+        if (config.isEnableTLS()) {
+            sb.https(config.getHttpsPort());
+            if (config.isEnableTlsSelfSigned()) {
+                sb.tlsSelfSigned();
+            } else {
+                try (InputStream cert =
+                             Files.newInputStream(Paths.get(config.getTlsCertChainPath())
+                                     .toFile().toPath());
+                     InputStream key = PrivateKeyUtil.loadDecryptionKey(config.getTlsKeyPath())) {
+                    sb.tls(cert, key);
```

Comment:
Https certificates are bound to a domain name so you have to add a configuration item to configure virtual host for that cert.